src/protagonist/API/Features/DeliveryChannels/DefaultDeliveryChannelRepository.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Collections.Generic;
+using System.IO.Enumeration;
+using DLCS.Core.Caching;
+using DLCS.Model.DeliveryChannels;
+using DLCS.Model.Policies;
+using DLCS.Repository;
+using LazyCache;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace API.Features.DeliveryChannels;
+
+public class DefaultDeliveryChannelRepository : IDefaultDeliveryChannelRepository
+{
+    private readonly IAppCache appCache;
+    private readonly CacheSettings cacheSettings;
+    private readonly ILogger<DefaultDeliveryChannelRepository> logger;
+    private readonly DlcsContext dlcsContext;
+
+    public DefaultDeliveryChannelRepository(
+        IAppCache appCache,
+        ILogger<DefaultDeliveryChannelRepository> logger,
+        IOptions<CacheSettings> cacheOptions,
+        DlcsContext dlcsContext)
+    {
+        this.appCache = appCache;
+        this.logger = logger;
+        cacheSettings = cacheOptions.Value;
+        this.dlcsContext = dlcsContext;
+    }
+    
+    public List<DefaultDeliveryChannel> GetDefaultDeliveryChannelsForCustomer(int customerId, int space)
+    {
+        var key = $"defaultDeliveryChannels:{customerId}";
+        
+        var defaultDeliveryChannels = appCache.GetOrAdd(key, () =>
+        {
+            logger.LogDebug("Refreshing {CacheKey} from database", key);
+
+            var defaultDeliveryChannels = dlcsContext.DefaultDeliveryChannels
+                .AsNoTracking()
+                .Include(d => d.DeliveryChannelPolicy)
+                .Where(d => d.Customer == customerId);
+
+            return defaultDeliveryChannels;
+        }, cacheSettings.GetMemoryCacheOptions(CacheDuration.Long));
+
+        return defaultDeliveryChannels.Where(d => d.Space == space || d.Space == 0).ToList();
+    }
+    
+    public List<DeliveryChannelPolicy> MatchedDeliveryChannels(string mediaType, int space, int customerId)
+    {
+        var completedMatch = new List<DeliveryChannelPolicy>();
+        
+        var defaultDeliveryChannels = GetDefaultDeliveryChannelsForCustomer(customerId, space);
+        
+        foreach (var defaultDeliveryChannel in defaultDeliveryChannels.OrderByDescending(v => v.Space).ThenByDescending(c => c.MediaType.Length))
+        {
+
+            if (completedMatch.Any(d => d.Channel == defaultDeliveryChannel.DeliveryChannelPolicy.Channel))
+            {
+                continue;
+            }
+
+            if (FileSystemName.MatchesSimpleExpression(defaultDeliveryChannel.MediaType, mediaType))
+            {
+                completedMatch.Add(defaultDeliveryChannel.DeliveryChannelPolicy);
+            }
+        }
+
+        return completedMatch;
+    }
+
+    public DeliveryChannelPolicy 
```

Comment:
This method shares some of the same logic as above method, it might be worth trying to encapsulate it in a single place to make for a single place to change. At the very least the ordering logic as that's key.

Also, formatting is off here - all other examples of methods with long param list will start to split params to next line rather than method name onto next line.

Reply:
I did think about encapsulating this, but thought the logic was different enough to keep it separate , but was still iffy about it - will see if I can make it better